### PR TITLE
Add LRSchedulerHook

### PR DESCRIPTION
**Pull request by thequilo**
Add a hook that applies a leanring rate scheduler to a training. This only supports schedulers that don't use metrics.

### CODE REVIEW

File: padertorch/train/hooks.py [hunk 1]
```diff
@@ -816,3 +817,43 @@ def close(self, trainer):
...
+    # It is very likely that this check is exclusive to this hook
+    IS_PYTORCH_1_1 = LooseVersion(torch.__version__) >= '1.1.0'
+
+    def __init__(self, lr_scheduler, trigger=(1, 'epoch')):
+        super().__init__(trigger)
+        self.lr_scheduler = lr_scheduler
+
+    def pre_step(self, trainer: 'pt.Trainer'):
+        if self.trigger(iteration=trainer.iteration, epoch=trainer.epoch):
+            if trainer.epoch > 0 or not self.IS_PYTORCH_1_1:
```
Comment (jensheit) [inline padertorch/train/hooks.py]: should the step be called if pytorch larger 1.1.0 or if it is not?

Comment (thequilo) [reply padertorch/train/hooks.py]: It should be skipped in the first epoch if pytorch is larger than 1.1.0

Comment (jensheit) [reply padertorch/train/hooks.py]: so it should be an and?

Comment (jensheit) [reply padertorch/train/hooks.py]: ah okay, to prevent the warning. Then it is fine

**Comment by boeddeker** [comment]
Can you move the code below the `BackOffValidationHook`? This hook also does some kind of learning rate scheduling.

**Comment by thequilo** [comment]
Yes, I'll open another pull request for it.

Btw, the file `hooks.py` is becoming quite large, might be worth thinking about how to split it.

**Comment by thequilo** [comment]
Or maybe I don't because I can just push to this repository.

File: padertorch/train/hooks.py (hunk 1 above)
Comment (thequilo) [reply padertorch/train/hooks.py]: It does not only prevent the warning. The learning rate schedule would be shifted by one epoch otherwise

File: padertorch/train/hooks.py [hunk 2]
```diff
@@ -816,3 +817,43 @@ def close(self, trainer):
...
+        ... ))  # doctest: +SKIP
+
+    Note:
+
+        This hook can only be used with learning rate schedulers that
+        don't require metrics.
+
+    """
+    # It is very likely that this check is exclusive to this hook
+    IS_PYTORCH_1_1 = LooseVersion(torch.__version__) >= '1.1.0'
```
Comment (boeddeker) [inline padertorch/train/hooks.py]: Can you change this to something like `PYTORCH_ge_1_1`? Do you have an issue, where they mentioned this shift?

Comment (thequilo) [reply padertorch/train/hooks.py]: Yes, and yes it's https://github.com/pytorch/pytorch/pull/7889

Comment (boeddeker) [reply padertorch/train/hooks.py]: Thanks, could you add that reference to the issue to the source code?

Comment (thequilo) [reply padertorch/train/hooks.py]: Done!

Comment (boeddeker) [reply padertorch/train/hooks.py]: Thanks, I found them directly in the master.
